presentation/src/main/java/org/sagebionetworks/research/presentation/model/form/InputFieldViewBase.java

```diff
@@ -88,8 +88,12 @@
     private final String uiHint;
 
     public static <E extends Comparable<E>> InputFieldViewBase<E> fromInputField(InputField<E> inputField,
-            DrawableMapper mapper) {
+            DrawableMapper mapper, @NonNull String formStepIdentifier) {
         String identifier = inputField.getIdentifier();
+        // If no identifier is provided in the InputField, use the form step's identifier
+        if (identifier == null) {
+            identifier = formStepIdentifier;
```

Comment:
Basically, yes. The input field identifier is always non-null, but as a part of decoding it, it checks the step identifier if there is none provided for the field. 

On iOS there is no "view" model class. Instead, we use data model, ViewModel, and ViewController (Fragment). We do not use PONSO objects in SageResearch since they really don't add any value (and as discussed before, calling them "View" objects is super-confusing. BridgeSDK uses PONSO objects but that's so that Erin can use KVO to write to the properties to/from both CoreData and JSON.